metadata-io/src/main/java/com/linkedin/metadata/entity/RetentionService.java

```diff
@@ -107,6 +107,7 @@ public boolean setRetention(@Nullable String entityName, @Nullable String aspect
     GenericAspect retentionAspect = GenericRecordUtils.serializeAspect(retentionConfig);
     aspectProposal.setAspect(retentionAspect);
     aspectProposal.setAspectName(Constants.DATAHUB_RETENTION_ASPECT);
+    aspectProposal.setChangeType(ChangeType.UPSERT);
```

Comment:
seems like we just need to set this

Reply:
was something breaking due to not having this? why didn't we see before

Reply:
This breaks a test, seems like if the change type is null it will fail a validation check. If non-timeseries, then this must be either PATCH or UPSERT. Null is no longer valid.